### DISCUSSION TITLE
Advanced User permissions / Membership

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#865] [Q&A]
Hi 

For know, I see that Mavo can handle logged-out(read only) or logged-in (edit by default) users,

So as soon as they login, they can edit.

1/ How can I handle this use case:  users can login only if they are members (with separated list of members?)

2/ Is it possible to manage a 'member' permission matrix? something like:

- admins can edit and delete
- loggin-in users can edit only
- logged-out users can read only

Is it possible with firebase or github plugins?

**Comment by LeaVerou** [comment]
Not sure I follow 1, if they are not logged in how do you know which member they are?

2 is touching on a big one, and actually a research direction that we may explore at some point in the future. See long discussion [here](https://github.com/mavoweb/mavo/discussions/720).

For specific cases like the one you describe, you *can* write out the permissions manually in Firebase and then hide the non-relevant controls with CSS. It's not great, but it will work.

**Comment by GalinhaLX** [comment]
1/ Sorry, let me try to explain better:

By default, mavo displays 'login' button
If user have a github account for example, he'll be able to edit the page, and save it to his github

But, let's say I want to restrain access to members (from a list), so:

A guest try to login with his Github account
Is he part of the list?
if yes: he's able to edit and save to his Github
if no: he can't login even if he has a github account (we can also propose to be a member)

It's some kind of pre-loggin process, where I need to check if users are members (declared in a list), then if they are members, i pass info to loggin trought 'loggin' button of Mavo

**Comment by GalinhaLX** [comment]
I guess it's better to do it Server Side

Checking that maybe Mavo PHP Plug in can help

or maybe an idea to do a MAVO permission plug-in that runs server side
